src/react-components/object-info-dialog.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import DialogContainer from "./dialog-container.js";
+import styles from "../assets/stylesheets/client-info-dialog.scss";
+import { FormattedMessage } from "react-intl";
+
+export default class ObjectInfoDialog extends Component {
+  static propTypes = {
+    scene: PropTypes.object,
+    el: PropTypes.object,
+    objectDisplayString: PropTypes.string,
+    onClose: PropTypes.func,
+    hubChannel: PropTypes.object
+  };
+
+  state = {
+    pinned: false
+  };
+
+  componentDidMount() {
+    this.updatePinnedState = this.updatePinnedState.bind(this);
+    this.pin = this.pin.bind(this);
+    this.unpin = this.unpin.bind(this);
+    this.props.scene.addEventListener("uninspect", this.props.onClose);
+    this.updatePinnedState();
+  }
+
+  updatePinnedState() {
+    this.setState({ pinned: this.props.el.components["networked"].data.persistent });
+  }
+
+  pin() {
+    if (!NAF.utils.isMine(this.props.el) && !NAF.utils.takeOwnership(this.props.el)) return;
+    this.props.el.emit("pinned", { el: this.props.el });
+    this.props.el.setAttribute("pinnable", "pinned", true);
```

Comment:
This line is suspicious. The line above will pin the entity, but the UI on `pin-networked-object-button`s will not update. Not sure which codepath is endorsed for pinning/unpinning these entities.